### ISSUE TITLE
Queuing tracks always starts playing immediately

### ISSUE THREAD

**Issue by chrishale** [spotify/ios-sdk#252]
I'm trying to queue some tracks ahead of time, then give the user the option to press a play button to start the playback.

The callback in `queueURIs:clearQueue:callback:` don't seem to fire quickly enough for me to use `setIsPlaying:callback:`, without a small blip of audio coming through (only when the track has music from 0:00, tracks with a fade in makes this unnoticeable)

I've even tried using `setVolume:callback:` and set it to 0 before queueing, but the blip of music still comes through - no idea how that could happen.

Considering there is a `queuePlay:` method anyway, it would be nice to have the option to play the queue manually.

**Comment by kevin--** [comment]
Agreed, pretty much all of the methods on the SPTStreamingController are unpredictable and will change your state. I implemented SPTTrackProvider protocol on my own object to finally gain control of this process.

**Comment by chrishale** [comment]
Yeah, I think skipNext kicks of playback again - which is a tad annoying.

How did you go about rebuilding your SPTTrackProvider protocol?

**Comment by kevin--** [comment]
Implement the methods for ```SPTTrackProvider``` (tracksForPlayback and playableUri). I returned nil from playableUri, then it will fetch the NSArray* from tracksForPlayback when you call
```SPTStreamingController playTrackProvider: callback:```

**Comment by possan** [comment]
Indeed, the playback logic is a bit messy which in turn generates some unexpected behaviour, we're working on a better api for the actual playback which will make this a bit more clear.

**Comment by dtrenz** [comment]
I'm trying to allow a user to queue up tracks when playback is paused, but every time they add a track, playback resumes due to this issue. Please fix. At the very least, it should be documented that queueURI methods have a side effect of starting/resuming playback.

**Comment by chrishale** [comment]
I'm still getting a small pop of noise every now and then, but I've tried a few things to circumvent this and have had some success with:

    self.player.setVolume(0) {
      (error: NSError!) in
      
      self.player.queueURIs(uris, clearQueue: true) {
        self.player.setIsPlaying(false, callback: nil)
        if self.player.currentPlaybackPosition < 5 {
          self.player.seekToOffset(0, callback: nil)
        }
      }
    }

Not a fix, but works for now!

**Comment by CVertex** [comment]
Is this queuing API still in progress? Or is there a way to do this with the current beta?

Thanks for a making a very nice API otherwise! I enjoy using it :smile:
